app/controllers/users.js

```diff
@@ -81,3 +103,53 @@ exports.listUsers = (req, res, next) => {
     })
     .catch(next);
 };
+
+exports.createAdminUser = (req, res, next) => {
+  const err = validateUserCreation(req.body);
+  if (err) return next(err);
+
+  const filter = {
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    email: req.body.email
+  };
+
+  userService
+    .getUsersByFilter(filter)
+    .then(users => {
+      const [user] = users;
+      if (user) {
+        user.permission = enums.PERMISSION.ADMINISTRATOR;
+        return user
+          .save()
+          .then(() => res.sendStatus(200))
+          .catch(error => {
+            throw errors.databaseError(error.message);
+          });
+      }
+
+      const emailFilter = {
```

Comment:
is this duplicated?

Reply:
I want a json object that only contains email as key and req.body.email as value. I don't think I can use filter.email to do that. Why do you think it is duplicated?